### DISCUSSION TITLE
DiffusionAnalyzer: initial and final structures do not match.

### DISCUSSION THREAD

**Discussion by thienbinh92** [materialsproject/pymatgen#2524] [Q&A] [answered]
Hello, I made a question in the [matsci.org](https://matsci.org/) community discussion forum last month, but I have not received any answer yet. I hope when I post a question here, someone may offer a favor.
I use DiffusionAnalyzer to analyze AIMD's results at 800, 1000, and 1200K:
`import matplotlib.pyplot as plt
import collections
from pymatgen.core import Structure
from pymatgen.analysis.diffusion_analyzer import DiffusionAnalyzer
file1 = ["test-400/vasprun.xml", "test-600/vasprun.xml", "test-800/vasprun.xml"] 
analyzer1 = DiffusionAnalyzer.from_files(file1, specie="Li", smoothed=False)`
I get ValueError: initial and final structures do not match. 
I do not really understand this error. Not sure if I make something wrong?

<img width="1055" alt="Screenshot 0004-05-17 at 23 22 05" src="https://user-images.githubusercontent.com/56110825/168834222-93c1f3a9-9866-4b1c-b458-89da3ba735ab.png">
++++++++++++++++++++++++++++++++++++
My procedure to perform AIMD is like this:
1. Run with the NPT ensemble with INCAR:
   ISIF   = 3
   TEBEG  = 1000
   TEEND  = 1000
   IBRION = 0 
   ISYM   = 0
   NSW    = 5000
   NBLOCK = 1
   MDALGO = 3
   LANGEVIN_GAMMA = 10 10 10 10
   LANGEVIN_GAMMA_L = 10
   SMASS  = 1.0 
   POTIM  = 2.0
2. Take CONTCAR with average volume, then copy it to POSCAR. Run with NVT ensemble:  ISIF   = 3 and NSW    = 25000. Then I use vasprun.xml and analyze the data with DiffusionAnalyzer.

**Comment by shyuep** [comment, marked as answer]
I am not sure what you are attempting to do, but you cannot analyze diffusivity using AIMD runs at three different temperatures. To calculate diffusivity at **one** temperature, you need a run with the same parameters, possibly spread across multiple start/stop runs (during which the initial structure of the next run must be identical to the final structure of the previous run). 

After you get the diffusivity at different temperatures, then you can create an Arrhenius plot. Please review the examples in the matgenb.materialsvirtuallab.org site for information.

**Reply by thienbinh92** [reply]
Thank you so much! I am clear now. 
I found in the example, that the MDS plot looks "spiky". Is it typical for raw data without a smooth option?

**Reply by shyuep** [reply]
Yes.
